### DISCUSSION TITLE
Languages supported?

### DISCUSSION THREAD

**Discussion by msr22** [mindee/doctr#837] [Q&A]
What all languages are supported by the OCR?

**Comment by fg-mindee** [comment]
Hello @msr22 :wave: 

I'll try to give you an exhaustive answer:
- for OCR specifically, the language per se isn't relevant but its vocabulary (list of supported characters) is, since it isn't trained to have a semantic understanding of the language.
- if you indeed mean vocabulary, as of now, our models are trained on the french vocab! You can find the list of vocabs over here :point_right:  https://github.com/mindee/doctr/blob/main/doctr/datasets/vocabs.py
- now we chose the french vocab, because it includes most european & american characters. So for instance, the english vocab is strictly included in the french one. Meaning that you can use this for any text with characters included in the french vocab :)

I hope this helps :+1:

**Reply by Yevgnen** [reply]
How can I add Chinese vocabulary support?

**Reply by felixdittrich92** [reply]
Hi @Yevgnen :wave:,
feel free to open a PR to add this in
https://github.com/mindee/doctr/blob/main/doctr/datasets/vocabs.py and updating in the docs https://github.com/mindee/doctr/blob/main/docs/source/modules/datasets.rst

**Reply by frgfm** [reply]
Oh yes please do @Yevgnen :pray:

**Comment by chaudhary-mohit** [comment]
Hey, As I can see there is no hindi language support. How I could achieve this using doctr to use for hindi as well?
@fg-mindee

**Reply by frgfm** [reply]
Hey there! So you'd need to add support of Hindi characters to the vocabs and at least train a small Hindi character classifier to check if the encoding works 👌

So sequentially:
- open a PR to add the Hindi characters to the vocab file
- then use our synthetic character classification pipeline using that vocab (i'm concerned we might have issues with the rendering part)
- if that works, we could give you easy directions to achieve your goal with limited computing resources 🙂

Let us know if you have questions!
